### PR TITLE
[Feat] Proposal to resolve the redirect url

### PR DESCRIPTION
**Pull request by rafaelsideguide**
Issue #311

### CODE REVIEW

**Comment by nickscamara** [comment]
@rafaelsideguide looks great. I think there are some conflicts. Once resolved, let's merge.

**Comment by rafaelsideguide** [comment]
@nickscamara conflicts solved

File: apps/api/src/scraper/WebScraper/index.ts [hunk 1]
```diff
@@ -209,7 +210,9 @@ export class WebScraperDataProvider {
   private async handleSingleUrlsMode(
     inProgress?: (progress: Progress) => void
   ): Promise<Document[]> {
-    const links = this.urls;
+    const links = await Promise.all(
```
Comment (nickscamara) [inline apps/api/src/scraper/WebScraper/index.ts]: remove from scrape

Comment (rafaelsideguide) [reply apps/api/src/scraper/WebScraper/index.ts]: done!

File: apps/api/src/scraper/WebScraper/crawler.ts [hunk 2]
```diff
@@ -134,6 +134,14 @@ export class WebCrawler {
     limit: number = 10000,
     maxDepth: number = 10
   ): Promise<{ url: string, html: string }[]> {
+    const initialUrl = this.initialUrl
+    this.initialUrl = await checksRedirect(this.initialUrl);
+    if (initialUrl != this.initialUrl) {
```
Comment (nickscamara) [inline apps/api/src/scraper/WebScraper/crawler.ts]: could this initialUrl be ever different from this.initialUrl due to http parsing etc?

**Review by nickscamara — COMMENTED** [review]
Added one more comment around the initialUrl mismatch.

File: apps/api/src/scraper/WebScraper/crawler.ts (hunk 2 above)
Comment (rafaelsideguide) [reply apps/api/src/scraper/WebScraper/crawler.ts]: @nickscamara I added "normalizeUrl"

**Review by nickscamara — APPROVED** [review]
lgtm

**Pull request closed without merging**